webapp/src/pages/boardPage/boardPage.tsx

```diff
@@ -215,13 +215,15 @@ const BoardPage = (props: Props): JSX.Element => {
                     UserSettings.setLastViewId(match.params.boardId, viewId)
                 }
             }
-
-            if (!props.readonly && me) {
-                loadOrJoinBoard(me.id, teamId, match.params.boardId)
-            }
         }
     }, [teamId, match.params.boardId, viewId, me?.id])
 
+    useEffect(() => {
+        if (match.params.boardId && !props.readonly && me) {
+            loadOrJoinBoard(me.id, teamId, match.params.boardId)
+        }
+    }, [teamId, match.params.boardId, me?.id])
+
```

Comment:
Could you explain the need for this change?

Reply:
This piece of code should only execute when `teamId`, `boardId` and `me.Id` changes, as it is dependent on it. But it should not execute whenever `viewId` changes.  That's the reason to separate it.